fee-provider/src/main/scala/org/bitcoins/feeprovider/AggregateFeeRateApi.scala

```diff
@@ -0,0 +1,34 @@
+package org.bitcoins.feeprovider
+
+import org.bitcoins.core.api.feeprovider.FeeRateApi
+import org.bitcoins.core.util.FutureUtil
+import org.bitcoins.core.wallet.fee.FeeUnit
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+case class AggregateFeeRateApi(providers: Vector[FeeRateApi])(implicit
+    ec: ExecutionContext)
+    extends FeeRateApi {
+
+  override def getFeeRate(): Future[FeeUnit] = {
+    val init: Option[FeeUnit] = None
+    val retOptF = FutureUtil.foldLeftAsync(init, providers) {
```

Comment:
Just be aware , this doesn't query all of them. I was looking for a better solution, i think we want something like `Future.foldUntilSuccess()` but that doesn't seem to exist :-(. 


There is a `Future.firstCompletedOf()` but that returns either failures or successes, not just succeses and i believe it starts all the Futures.

Reply:
That is kinda what this is doing it only attempts the fee rate api until it has a success.

Reply:
Ahh ok. I see. That makes sense to me.